crates/protocol/src/package/state.rs

```diff
@@ -386,6 +386,12 @@ impl LogState {
             return Err(ValidationError::InitialEntryAfterBeginning);
         }
 
+        if signer_key_id != &init_key.fingerprint() {
+            return Err(ValidationError::KeyIDNotRecognized {
+                key_id: signer_key_id.clone(),
+            });
+        }
```

Comment:
This requires that the public key in the init entry (which is effectively a "grant") matches the signing key. Not strictly necessary for this PR but seems like a good conservative check for now.